x-pack/plugins/triggers_actions_ui/public/application/type_registry.ts

```diff
@@ -41,6 +41,30 @@ export class TypeRegistry<T extends BaseObjectType> {
     this.objectTypes.set(objectType.id, objectType);
   }
 
+  /**
+   * Registers async an object type to the type registry
+   */
+  public async registerAsync(getObjectType: () => Promise<T>) {
+    try {
+      const objectType = await getObjectType();
+      if (!this.has(objectType.id)) {
```

Comment:
I think if we're adding this new function, the behavior should be the same as `register` if `this.has(objectType.id)` which is to throw an error, not just silently do nothing.